### DISCUSSION TITLE
Question about updating to v3 patcher

### DISCUSSION THREAD

**Discussion by N-R-K** [ryanoasis/nerd-fonts#1203] [Q&A]
Hi,

I have a question regarding updating to nerd font v3. My situation in short:

* I have some major problems with newer versions of Jetbrains Mono so I've been using an older version.
* However, I've been keeping it updated with newer nerd font patcher manually.
* So far, I've been simply applying new versions of patcher to the previously patched font.

But since v3 has breaking changes, my question is what is the recommended way to update. Should I run v3 patcher on top of the font that was already patched (with v2 patcher) or should I run it on top of a "clean" font?

**Comment by Finii** [comment]
I would (always) start with the old but unpatched JetBrains Mono font and do not re-patch.

The reason for that is that older font-patchers changed some properties in the fonts that they should not, and you can not restore that values by repatching, they are gone.

But maybe the problems you have with newer fonts is just that - you liked some global font property changes introduced by the patching.

In principle you can always patch over patched fonts. Apart from the naming maybe. I guess the result would be a `JetBrainsMono Nerd Font Nerd Font`, because `Nerd Font` is appended each time you patch.

The 'breaking changes' themselves are no problem.

### The naming change

This is probably irrelevant for you, because you install the font manually and changing the font in your terminal settings is not a big problem. If you do not like the new naming you can specify `--makegroups 0` while patching and the old naming algorithm will be used. Well, I did not test if that still works, but at least I think I did not touch it.
I would not suggest that, esp for JetBrains Mono which was broken font-family wise with the old naming - if you install all weights. But maybe you are not doing that.

### The new Material Design Icons

If you patch over a previously patched font you will not loose the old Material Design Icons - mostly. Some will be overwritten by the expanding Octicons. Compare the table [here](https://github.com/ryanoasis/nerd-fonts/wiki/Glyph-Sets-and-Code-Points#overview)
If you do a fresh patch on a clean source there will be no old Material Design Icons, just the new ones at the new codepoints.

If you really need the old Material Design Icons but would like to clean-patch, you can edit the font-patcher and enable the old symbols (they are still there, at least at the moment), just change the value from `False` to `True`

![image](https://user-images.githubusercontent.com/16012374/236395562-b5bb150a-8909-47dc-83bf-081dd638820d.png)

### Box drawing glyphs

The box drawing glyphs are only patched in by font-patcher if the sourcefont does not have a complete set. So once the font-patcher patched in the box drawing glyphs they will not be touched by subsequent font-patcher runs - if their scaling/look/whatever is changed in some font-patcher revision you will not see it.
But the box drawing glyphs are a more recent addition, maybe your already patched font does not have them (or your JetBrains Mono had a complete set anyhow, then font-patcher will not add anything). So I would not expect a problem here this time.

**Reply by Finii** [reply]
About the naming selection...
```python
    # --makegroup has an additional undocumented numeric specifier. '--makegroup' is in fact '--makegroup 1'.
    # Original font name: Hugo Sans Mono ExtraCondensed Light Italic
    #                                                             NF  Fam agg.   
    # 0  turned off, use old naming scheme                        [-] [-] [-]
    # 1  HugoSansMono Nerd Font ExtraCondensed Light Italic       [ ] [ ] [ ]
    # 2  HugoSansMono Nerd Font ExtCn Light Italic                [ ] [X] [ ]
    # 3  HugoSansMono Nerd Font XCn Lt It                         [ ] [X] [X]
    # 4  HugoSansMono NF ExtraCondensed Light Italic              [X] [ ] [ ]
    # 5  HugoSansMono NF ExtCn Light Italic                       [X] [X] [ ]
    # 6  HugoSansMono NF XCn Lt It                                [X] [X] [X]
```

The prepatched fonts in the release use this:

```
$ cat src/unpatched-fonts/JetBrainsMono/config.cfg
config_patch_flags="--makegroups 4"
```

**Reply by N-R-K** [reply]
Thanks for the detailed answer!

> The reason for that is that older font-patchers changed some properties in the fonts that they should not

Interesting.. Can you elaborate a bit more on what properties exactly were being changed (or point me towards an existing issue/discussion) ?

The reason I'm curious about this is because one thing that drove me a bit nuts yesterday was when I downloaded a clean (but still old/same) version of JetBrainsMono and noticed that my patched version is "wider" compared to the clean version.

![image](https://user-images.githubusercontent.com/79544946/236450224-bd606ac1-eb00-4376-bf3d-44e47de6a138.png)

Both of these fonts are JetBrainsMono v2.0.1. But the top is patched (and re-patched) by font-patcher while the bottom one is a clean version. And as you can see, the top one is clearly "wider" than the bottom one.

I'm wondering if this is something that's caused by the older font patcher or not (I like the wider variant).
